aea/aea_builder.py

```diff
@@ -1186,20 +1192,26 @@ def _build_identity_from_wallet(self, wallet: Wallet) -> Identity:
         if self._name is None:  # pragma: nocover
             raise ValueError("You must set the name of the agent.")
 
+        default_ledger = self.get_default_ledger_not_null()
         if not wallet.addresses:
-            raise ValueError("Wallet has no addresses.")
+            raise AEAWalletNoAddressException("Wallet has no addresses.")
+
+        if default_ledger not in wallet.addresses:
+            raise ValueError(
+                f"Specified default ledger '{default_ledger}' not found in available addresses of types: {'[' + ','.join(wallet.addresses.keys()) + ']'}"
+            )
```

Comment:
without this, we would have a cryptic error message like
```
Error: 'fetchai'
```
typical of `ValueError` when a key is not present in a dictionary (see below `wallet.addresses[default_ledger]`)

Reply:
<img width="631" alt="Screenshot 2021-03-21 at 12 22 03" src="https://user-images.githubusercontent.com/19314578/111904693-17658000-8a40-11eb-8a4c-4dfa430b376b.png">
On current develop ^

Reply:
So this error message already exists somewhere.

Reply:
What needs fixing is to tell the user to run `aea install`!

Reply:
no, these are two different issues. Assume all the plug-ins are installed correctly.

1) `default_ledger` specified, NO addresses in the wallet:
you get `Wallet has no addresses`, which doesn't give suggestion on how to solve the problem (add key, generate key)
2) `default_ledger` specified, some addresses in the wallet but NONE of them of type `default_ledger`
you get `Error: 'fetchai'`, and the commented change fixes that by explaining what happened.

Reply:
if default_ledger would allow null values, we would have another case that would stop the build of the AEA.

Reply:
1. Extend error message looks like the solution.
2. Yes a helpful error message is needed there.
Avoid null value of default ledger, not helpful.